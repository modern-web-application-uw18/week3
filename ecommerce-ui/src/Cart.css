.Cart {
    font-size: 1em;
    color: Grey;
    align: left;
    margin: 20px;

}

.Cart__button {
    border: none;
    font-size: 2em;
    padding: 10px 30px;
    margin: 10px;
    cursor: pointer;
}

.Cart__button--addroom {
    align: right;
    color: red;
}

.Cart__button--removeroom {
    background-color: red;
}

.Cart__button--clear {
    

}