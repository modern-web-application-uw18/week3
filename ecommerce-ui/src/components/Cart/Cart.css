.Cart {
    background-color: rgb(95, 95, 95);
    height: 50px;
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    font-style: italic;
    font-size: 40px;
    color: #ffffff;
    margin-bottom: 0;
    text-align: right;

}


@media (max-width: 650px) {
    .Cart {
        font-size: 30px;
    }
}