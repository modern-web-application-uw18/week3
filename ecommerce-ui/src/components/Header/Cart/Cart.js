import React from 'react';
import Modal from 'react-bootstrap/lib/Modal';
import Button from 'react-bootstrap/lib/Button';
import MdHighlightRemove from 'react-icons/lib/md/highlight-remove';
import PropTypes from 'prop-types';



const cart = (props) => (
    

    <Modal show={props.cartShow} onHide={props.handleCartClose}>
        <Modal.Header closeButton>
            <Modal.Title>Shopping Cart</Modal.Title>
        </Modal.Header>
        <Modal.Body>


   

            <table className="table table-hover">
                <thead>
                    <tr>
                        <th></th>
                        <th scope="col">Title</th>
                        <th scope="col">City</th>
                        <th scope="col">Country</th>
                        <th scope="col">Cost</th>
                    </tr>
                </thead>
                <tbody>


                    {props.cartContent.map((item, idx) => {
                        return (
                        <tr key={idx}>
                            <td><MdHighlightRemove style={{color:'red'}} onClick={()=>props.removeListingFromTheCartHandler(item)}/></td>
                            <td>{item.title}</td>
                            <td>{item.location.city}</td>
                            <td>{item.location.country}</td>
                            <td>$ {item.payment.cost}</td>
                        
                        </tr>)
                    })}

                    <tr>
                        <td/>
                        <td/>
                        <td/>
                        <td><b>Total</b></td>
                        <td>$ {props.cartContent.reduce((sum,value) => sum+value.payment.cost, 0)}</td>
                    </tr>


                </tbody>
            </table>


        </Modal.Body>
        <Modal.Footer>
            <Button onClick={props.handleCartClose}>Close</Button>
        </Modal.Footer>
    </Modal>
)


cart.propTypes = {
    cartShow: PropTypes.bool.isRequired,
    handleCartClose: PropTypes.func.isRequired,
    cartContent: PropTypes.array.isRequired,
    removeListingFromTheCartHandler: PropTypes.func.isRequired
}


export default cart;
