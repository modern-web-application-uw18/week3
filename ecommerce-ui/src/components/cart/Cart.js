import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './cart.css';

class Cart extends Component {
  render() {

    const { length } = this.props.inCart;

    const cartTotal = () => {
      let total = 0;
      for (let rental of this.props.inCart) {
        total += rental.payment.cost
      }

      return total;
    }

    return (
      <aside className="cart-container"  id="cart-title">
        <h2 className="cart-container__title">Requested Rentals</h2>

        {length === 0 &&
          <div className="cart-item__total">You have no selections</div>
        }

        <div className="cart-item__container">
          {this.props.inCart.map((rental, index) =>
            <div key={index} className="cart-item">
              <h3 className="cart-item__title">{rental.title}</h3>
              <p className="cart-item__cost">${rental.payment.cost}</p>
              <button className="btn cart-item__btn" onClick={this.props.deleteRental(index)}>Remove</button>
            </div>
          )}

          {length !== 0 &&
            <div>
              <h2 className="cart-container__title">Request Total</h2>
              <div className="cart-item__total">
                {length} rental{length === 1 ? '' : 's'} requested | <strong>Total ${cartTotal()}</strong>
              </div>
            </div>
          }

        </div>
      </aside>
    );
  }
}

Cart.propTypes = {
  deleteRental: PropTypes.func.isRequired
}

export default Cart;
