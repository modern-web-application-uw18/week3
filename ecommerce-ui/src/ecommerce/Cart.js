import React, { Component } from 'react';
import PropTypes from 'prop-types';
import CartItem from './CartItem';

class Cart extends Component {

 render() {
   return (
       <form>
         <div>
           <h2>Properties in your cart:</h2>
            {this.props.itemsInCart.map((airbnb, idx) =>
             <CartItem
             removeFromCart={this.props.removeFromCart(idx)}
             key={idx}
             title={airbnb.title}
             cost={airbnb.payment.cost} />
            )}
           <div>
             <h1>Cart total: {'$' + this.props.getCartTotal()}</h1>
           </div>
        </div>
      </form>
   );
 }
}

Cart.propTypes = {
 removeFromCart: PropTypes.func,
 itemsInCart: PropTypes.array,
 getCartTotal: PropTypes.func
};

export default Cart;
