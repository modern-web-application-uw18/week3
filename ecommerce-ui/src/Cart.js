import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './Cart.css';

class Cart extends Component {

    constructor(props) {
        super(props);
        this.removeRoomAction = this.props.removeRoomAction.bind(this);
      }
    
    render() {
        let removeRoomAction = this.removeRoomAction;
        return (
            // <div className="Cart">
            //     <h2>{this.state.value}</h2>
            //     <button className="Cart____button--removeroom" onClick={this.decrement}>Remove Room</button>
            //     <button className="Cart__button--addroom" onClick={this.increment}>Add Room</button>
            // </div>
            <div class="Cart">
            <h1>Saved Rooms</h1>
            <ul>
            {this.props.rooms.map(function(room, index){
                return (<li key={ index }>{room.title}&nbsp;&nbsp;${room.payment.cost}&nbsp;&nbsp;<button type="button" onClick={() => removeRoomAction(room)}>Delete Room</button></li>);
              })}
            </ul>
            </div>
        );
    }
}

Cart.propTypes = {
    removeRoomAction: PropTypes.func,
    rooms: PropTypes.array
};

Cart.defaultProps = {
    rooms: []
};

export default Cart;