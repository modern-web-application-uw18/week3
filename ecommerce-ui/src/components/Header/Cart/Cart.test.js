import React from 'react';
import { shallow, mount} from 'enzyme';
import Cart from './Cart';
import Modal from 'react-bootstrap/lib/Modal';
import Button from 'react-bootstrap/lib/Button';
import MdHighlightRemove from 'react-icons/lib/md/highlight-remove';

describe('<Cart />', () => {

    const cartContent = [
        {
            "title": "Centrally-located Manhattan studio",
            "houseType": "Apartment",
            "image": "https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80",
            "location": {
                "city": "New York",
                "country": "USA"
            },
            "payment": {
                "cost": 105,
                "description": "Free cancellation"
            },
            "host": {
                "name": "Robert",
                "isSuperhost": true
            },
            "rating": {
                "stars": 3,
                "reviews": 133
            }
        },
    ];

    it('Snapshot test', () => {
        const wrapper = mount(<Cart cartContent={cartContent} />);
        expect(wrapper).toMatchSnapshot();
    })
})