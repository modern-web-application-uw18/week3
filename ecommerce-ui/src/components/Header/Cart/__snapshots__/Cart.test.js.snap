// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cart /> Snapshot test 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Shopping Cart
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <table
      className="table table-hover"
    >
      <thead>
        <tr>
          <th />
          <th
            scope="col"
          >
            Title
          </th>
          <th
            scope="col"
          >
            City
          </th>
          <th
            scope="col"
          >
            Country
          </th>
          <th
            scope="col"
          >
            Cost
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          key="0"
        >
          <td>
            <MdHighlightRemove
              onClick={[Function]}
              style={
                Object {
                  "color": "red",
                }
              }
            />
          </td>
          <td>
            Centrally-located Manhattan studio
          </td>
          <td>
            New York
          </td>
          <td>
            USA
          </td>
          <td>
            $ 
            105
          </td>
        </tr>
        <tr
          key="1"
        >
          <td>
            <MdHighlightRemove
              onClick={[Function]}
              style={
                Object {
                  "color": "red",
                }
              }
            />
          </td>
          <td>
            Denver apartment outside Union Station
          </td>
          <td>
            Denver
          </td>
          <td>
            USA
          </td>
          <td>
            $ 
            130
          </td>
        </tr>
        <tr
          key="2"
        >
          <td>
            <MdHighlightRemove
              onClick={[Function]}
              style={
                Object {
                  "color": "red",
                }
              }
            />
          </td>
          <td>
            Quaint 3-bedroom home in Gamla stan
          </td>
          <td>
            Stockholm
          </td>
          <td>
            Sweden
          </td>
          <td>
            $ 
            230
          </td>
        </tr>
        <tr>
          <td />
          <td />
          <td />
          <td>
            <b>
              Total
            </b>
          </td>
          <td>
            $ 
            465
          </td>
        </tr>
      </tbody>
    </table>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      Close
    </Button>
  </ModalFooter>
</Modal>
`;
